src/main/java/com/zendesk/maxwell/bootstrap/AsynchronousBootstrapper.java

```diff
@@ -0,0 +1,140 @@
+package com.zendesk.maxwell.bootstrap;
+
+import com.zendesk.maxwell.*;
+import com.zendesk.maxwell.producer.AbstractProducer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.sql.SQLException;
+import java.util.LinkedList;
+import java.util.Queue;
+
+public class AsynchronousBootstrapper extends AbstractBootstrapper {
+
+	static final Logger LOGGER = LoggerFactory.getLogger(MaxwellReplicator.class);
+
+	private Thread thread = null;
+	private Queue<RowMap > queue = new LinkedList<>();
+	private RowMap bootstrappedRow = null;
+	private RowMapBufferByTable skippedRows = null;
+	private SynchronousBootstrapper synchronousBootstrapper = getSynchronousBootstrapper();
+
+	public AsynchronousBootstrapper( MaxwellContext context ) throws IOException {
+		super(context);
+		skippedRows = new RowMapBufferByTable();
+	}
+
+	@Override
+	public boolean isStartBootstrapRow(RowMap row) {
+		return synchronousBootstrapper.isStartBootstrapRow(row);
+	}
+
+	@Override
+	public boolean isCompleteBootstrapRow(RowMap row) {
+		return synchronousBootstrapper.isCompleteBootstrapRow(row);
+	}
+
+	@Override
+	public boolean isBootstrapRow(RowMap row) {
+		return synchronousBootstrapper.isBootstrapRow(row);
+	}
+
+	protected SynchronousBootstrapper getSynchronousBootstrapper( ) {
+		return new SynchronousBootstrapper(context);
+	}
+
+	@Override
+	public boolean shouldSkip(RowMap row) throws SQLException, IOException {
+		// The main replication thread skips rows of the currently bootstrapped
```

Comment:
I'm trying to understand the async bootstrapper and how it interacts with the kafka checkpoints. 

I understand the part where rows from bootstrapped tables are saved into memory as the bootstrap is happening and then are emitted.

When rows from non-bootstrapped tables are sent to the producer, won't the producer checkpoint those rows? The checkpoint for those rows might be beyond the point of the "maxwell.bootstrap" row, and upon crash, they would not be restarted.

And it looks like when you send the INSERTs of a bootstrap, won't the producer.send() end up checkpointing the rows for those INSERTS?